src/fns.rs

```diff
@@ -747,72 +747,105 @@ pub(crate) fn string_bytes(string: &str) -> i64 {
 }
 
 #[defun]
-pub(crate) fn string_lessp(string1: &str, string2: &str) -> bool {
-    for (c1, c2) in string1.chars().zip(string2.chars()) {
+pub(crate) fn string_lessp<'ob>(string1: Object<'ob>, string2: Object<'ob>) -> Result<bool> {
+    let string1 = match string1.untag() {
+        ObjectType::String(x) => x.chars().collect::<Vec<_>>(),
+        ObjectType::Symbol(x) => x.as_str().chars().collect::<Vec<_>>(),
+        _ => bail!(TypeError::new(Type::String, string1)),
+    };
+    let string2 = match string2.untag() {
+        ObjectType::String(x) => x.chars().collect::<Vec<_>>(),
+        ObjectType::Symbol(x) => x.as_str().chars().collect::<Vec<_>>(),
+        _ => bail!(TypeError::new(Type::String, string2)),
+    };
```

Comment:
It looks like you were fighting the borrow checker here. The reason is that the auto-deref of `Symbol` (to call `as_str()` on `SymbolCellInner`) won't let the `&str` live past the match scope because `Symbol` will be dropped. If you refactor it like this using the `get()` method you can avoid that issue and not need to create these temp vectors.

```rust
    let string1 = match string1.untag() {
        ObjectType::String(x) => x,
        ObjectType::Symbol(x) => x.get().name(),
        _ => bail!(TypeError::new(Type::String, string1)),
    };
    let string2 = match string2.untag() {
        ObjectType::String(x) => x,
        ObjectType::Symbol(x) => x.get().name(),
        _ => bail!(TypeError::new(Type::String, string2)),
    };
```

Reply:
Thank you.